api/v3/Payment.php

```diff
@@ -41,6 +41,9 @@
  *   Array of financial transactions which are payments, if error an array with an error id and error message
  */
 function civicrm_api3_payment_get($params) {
+  if (!CRM_Core_Permission::check('access CiviContribute')) {
```

Comment:
(further clarification - in php you can choose what you want check_permissions to be by setting it ie 
$params['check_permissions'] => TRUE;

In AJAX it will always be set to TRUE - you can't override it from AJAX